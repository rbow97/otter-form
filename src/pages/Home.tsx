import React, { useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faHeart } from '@fortawesome/free-solid-svg-icons'
import Form from '../components/Form/Form';
import otter from '../images/otter.svg'

const Home: React.FC = () => {
    const [open, setOpen] = useState(false);

    return (
        <>
            <div className="header">
                <h1>Otter</h1>
            </div>
            <div className='content'>
                <img className='content-image' src={otter} alt='otter' />
                <span className='content-title'>The new way to happiness is through Otter!</span>
                <span className='content-subtitle'>Be the first to know when we launch</span>
                <button onClick={() => setOpen(true)}>Request an invite</button>
            </div>
            <Form open={open} setOpen={setOpen} />
            <div className="footer">
                <div className='footer-text'>
                    <span>Made with</span>
                    <FontAwesomeIcon color='#ff69b4' icon={faHeart} />
                    <span>in London</span>
                </div>
                <span>Copywrite 2020 Otter Limited</span>
            </div>
        </>
    )
}

export default Home
