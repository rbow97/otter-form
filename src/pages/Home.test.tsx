import { render } from '@testing-library/react';
import React from 'react';
import Home from './Home';

describe('Home', () => {
  it('should match snapshot', () => {
    const {container} = render(
      <Home />
    );
    expect(container).toMatchSnapshot();
  });
  it('should render without error', () => {
      const renderedText = 'Copywrite 2020 Otter Limited'
    const {container} = render(
      <Home/>
    )
    expect(container).toBeInstanceOf(HTMLElement);
    expect(renderedText).toBeTruthy();
  });
})
