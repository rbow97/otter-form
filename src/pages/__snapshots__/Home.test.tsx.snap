// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<div>
  <div
    class="header"
  >
    <h1>
      Otter
    </h1>
  </div>
  <div
    class="content"
  >
    <img
      alt="otter"
      class="content-image"
      src="otter.svg"
    />
    <span
      class="content-title"
    >
      The new way to happiness is through Otter!
    </span>
    <span
      class="content-subtitle"
    >
      Be the first to know when we launch
    </span>
    <button>
      Request an invite
    </button>
  </div>
  <div
    class="footer"
  >
    <div
      class="footer-text"
    >
      <span>
        Made with
      </span>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-heart fa-w-16 "
        color="#ff69b4"
        data-icon="heart"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
          fill="currentColor"
        />
      </svg>
      <span>
        in London
      </span>
    </div>
    <span>
      Copywrite 2020 Otter Limited
    </span>
  </div>
</div>
`;
